tests/Api/Shop/Checkout/PaymentMethodTest.php

```diff
@@ -0,0 +1,299 @@
+            'channel.yaml',
+            'cart.yaml',
+            'country.yaml',
+            'shipping_method.yaml',
+            'payment_method.yaml',
+        ]);
+
+        /** @var PaymentMethodInterface $paymentMethod */
+        $paymentMethod = $fixtures['payment_method_bank_transfer'];
+        /** @var CustomerInterface $customer */
+        $customer = $fixtures['customer_oliver'];
+
+        $tokenValue = $this->pickUpCart(email: $customer->getEmailCanonical());
+        $this->addItemToCart('MUG_BLUE', 3, $tokenValue);
+        $cart = $this->updateCartWithAddress($tokenValue, $customer->getEmailCanonical());
+        $cart = $this->dispatchShippingMethodChooseCommand($tokenValue, 'DHL', $cart->getShipments()->first()->getId());
+
+        $this->client->request(
+            method: 'PATCH',
+            uri: sprintf('/api/v2/shop/orders/%s/payments/%s', $tokenValue, $cart->getLastPayment()->getId()),
+            server: $this->headerBuilder()->withMergePatchJsonContentType()->withJsonLdAccept()->withShopUserAuthorization($customer->getEmailCanonical())->build(),
+            content: json_encode([
+                'paymentMethod' => $paymentMethod->getCode(),
+            ], \JSON_THROW_ON_ERROR),
+        );
+
+        $this->assertResponse(
+            $this->client->getResponse(),
+            'shop/checkout/payment_method/select_payment_method_by_user',
+        );
+    }
+
+    /** @test */
+    public function it_does_not_select_payment_method_of_order_created_by_a_user_authenticated_as_another_user(): void
+    {
+        $fixtures = $this->loadFixturesFromFiles([
+            'authentication/shop_user.yaml',
+            'channel.yaml',
+            'cart.yaml',
+            'country.yaml',
+            'shipping_method.yaml',
+            'payment_method.yaml',
+        ]);
+
+        /** @var PaymentMethodInterface $paymentMethod */
+        $paymentMethod = $fixtures['payment_method_bank_transfer'];
+        /** @var CustomerInterface $customer */
+        $customer = $fixtures['customer_oliver'];
+
+        $tokenValue = $this->pickUpCart(email: $customer->getEmailCanonical());
+        $this->addItemToCart('MUG_BLUE', 3, $tokenValue);
+        $cart = $this->updateCartWithAddress($tokenValue, $customer->getEmailCanonical());
+        $cart = $this->dispatchShippingMethodChooseCommand($tokenValue, 'DHL', $cart->getShipments()->first()->getId());
+
+        $this->client->request(
+            method: 'PATCH',
+            uri: sprintf('/api/v2/shop/orders/%s/payments/%s', $tokenValue, $cart->getLastPayment()->getId()),
+            server: $this->headerBuilder()->withMergePatchJsonContentType()->withJsonLdAccept()->withShopUserAuthorization('dave@doe.com')->build(),
+            content: json_encode([
+                'paymentMethod' => $paymentMethod->getCode(),
+            ], \JSON_THROW_ON_ERROR),
+        );
+
+        $this->assertResponseCode($this->client->getResponse(), Response::HTTP_NOT_FOUND);
```

Comment:
```suggestion
        $this->assertResponseCode($this->client->getResponse(), Response::HTTP_FORBIDDEN);
```
Just opening a discussion. In another test, in a similar case where one user tries to retrieve another user's resource, we return a forbidden response, which makes sense. It would be good to take the same approach for all these similar cases.

Reply:
In your case, as a guest, if you put in the wrong token, you get a NOT_FOUND error. However, if you enter the correct token, you get a FORBIDDEN error, which reveals that the token exists in the database.